### PR TITLE
Fix nightly clippy errors

### PR DESCRIPTION
**Pull request by cmichi**

### CODE REVIEW

File: crates/env/src/call/call_builder.rs [hunk 1]
```diff
@@ -30,6 +30,7 @@ use core::marker::PhantomData;
 
 /// The final parameters to the cross-contract call.
 #[derive(Debug)]
+#[allow(dead_code)]
```
Comment (cmichi) [inline crates/env/src/call/call_builder.rs:33]: Not really happy with the solution of adding the `#[allow(dead_code)]` attributes, any other ideas?

Comment (Robbepop) [reply crates/env/src/call/call_builder.rs:33]: how about some proper `#[cfg(..)]`?

Comment (Robbepop) [reply crates/env/src/call/call_builder.rs:33]: I'd not like to accept an `#[allow(dead_code)]` hack-fix.

Comment (cmichi) [reply crates/env/src/call/call_builder.rs:33]: > how about some proper `#[cfg(..)]`?

The `call_builder` is only used in examples and tests, but not in the crate itself ‒ that's why the fields are never read there. How would a `#[cfg(…)]` fix this? We run `clippy` on all targets.

File: crates/env/src/call/call_builder.rs [hunk 2]
```diff
@@ -41,17 +41,11 @@ where
     /// The transferred value for the call.
     transferred_value: E::Balance,
     /// The expected return type.
-    return_type: ReturnType<R>,
+    _return_type: ReturnType<R>,
```
[the remark this answers is not in the document]
Comment (cmichi) [reply crates/env/src/call/call_builder.rs:44]: I think it's used to preserve the `R`, so kind of like `PhantomData`?

**Review by Robbepop — APPROVED** [review]
LGTM

**Pull request merged**